syntax = "proto3";

enum OperationType {
  CREATE = 0;
  UPDATE = 1;
  DELETE = 2;
}


message CRDTOperation {
  string file_id = 1;
  OperationType operation_type = 2;
  bytes payload = 3;
  uint64 logical_timestamp = 4;
  string peer_id = 5;
  string parent_hash = 6;
}
